controllers/chat/moveToAnon.js

```diff
@@ -0,0 +1,130 @@
+const {User, ChatAnonUserInfo} = require('../../databases/models');
+const {StreamChat} = require('stream-chat');
+const ErrorResponse = require('../../utils/response/ErrorResponse');
+const {responseError, responseSuccess} = require('../../utils/Responses');
+const {CHANNEL_TYPE} = require('../../helpers/constants');
+const UsersFunction = require('../../databases/functions/users');
+
+/**
+ *
+ * @param {import("express").Request} req
+ * @param {import("express").Response} res
+ */
+const moveToAnon = async (req, res) => {
+  const {
+    targetUser,
+    oldChannelId,
+    anon_user_info_color_code,
+    anon_user_info_color_name,
+    anon_user_info_emoji_code,
+    anon_user_info_emoji_name
+  } = req.body;
+
+  let members = [targetUser];
+  if (!members.includes(req.userId)) members.push(req.userId);
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This code assumes that `targetUser` is always an array. If a single user is passed, this will throw an error. You should add a check to ensure `targetUser` is an array before proceeding.
```diff
- let members = [targetUser];
+ let members = Array.isArray(targetUser) ? targetUser : [targetUser];
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->